### PR TITLE
Fix shortcuts on My page, Version, Work Package, Meetings etc.

### PR DESCRIPTION
**Pull request by knight-bubble**
[`* `#19106` Editing page with "g" + "e" not possible for several pages`](http://community.openproject.org/work_packages/19106)
Now it will work stable

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hirtie-maxim-fix-19106-editing-pages-with-access-keys.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hirtie-maxim-fix-19106-editing-pages-with-access-keys.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/views/my/page.html.erb [hunk 1]
```diff
@@ -28,7 +28,7 @@ See doc/COPYRIGHT.rdoc for more details.
 ++#%>
 <% breadcrumb_paths(l(:label_my_page)) %>
 <% content_for :action_menu_specific do %>
-  <%= link_to l(:label_personalize_page), action: 'page_layout', accesskey: accesskey(:edit) %>
+  <%= link_to l(:label_personalize_page), { :action => 'page_layout', :accesskey => accesskey(:edit) }, "accesskey" => "3" %>
```
Comment (myabc) [inline app/views/my/page.html.erb]: @hirtie-maxim why did you change the syntax here?

Comment (knight-bubble) [inline app/views/my/page.html.erb]: @myabc, I had to add the accesskey attribute to link.

Comment (floriank) [inline app/views/my/page.html.erb]: that does not require using the old hash syntax again :laughing:

**Comment by floriank** [comment]
@myabc Is this the usual flickering? @hirtie-maxim If it's not the usual flickering, could you check?

**Comment by myabc** [comment]
@floriank .. the cukes, yes. Karma + Protractor, no.

**Comment by NobodysNightmare** [comment]
hmm... got the false impression that this PR should be assigned to Maxim, but it looks fine. Will start reviewing...

File: app/views/my/page.html.erb [hunk 2]
```diff
@@ -28,7 +28,7 @@ See doc/COPYRIGHT.rdoc for more details.
 ++#%>
 <% breadcrumb_paths(l(:label_my_page)) %>
 <% content_for :action_menu_specific do %>
-  <%= link_to l(:label_personalize_page), action: 'page_layout', accesskey: accesskey(:edit) %>
+  <%= link_to l(:label_personalize_page), { :action => 'page_layout', :accesskey => accesskey(:edit) }, "accesskey" => accesskey(:edit) %>
```
Comment (NobodysNightmare) [inline app/views/my/page.html.erb:31]: Why did you replace the new hash-syntax with the old one?

new: `foo: 'bar'`
old: `:foo => 'bar'`

Comment (myabc) [inline app/views/my/page.html.erb:31]: `:accesskey => accesskey(:edit)` doesn't belong in the first Hash.

Comment (NobodysNightmare) [inline app/views/my/page.html.erb:31]: specifying access key twice, once in the hash and ocne outside (what does that mean anyway?) looks suspicious...

Speculation: ~~Either this change is not needed~~ or the source of the problem lies elsewhere...

edit: Tried it... this change is needed, don't understand it, though...

File: frontend/app/services/keyboard-shortcut-service.js [hunk 3]
```diff
@@ -93,6 +95,26 @@ module.exports = function($window, $rootScope, $timeout, PathHelper) {
     };
   }
 
+  function clickLink(link) {
+    var cancelled = false;
+
+    if (document.createEvent) {
+        var event = document.createEvent('MouseEvents');
+        event.initMouseEvent('click', true, true, window,
```
Comment (myabc) [inline frontend/app/services/keyboard-shortcut-service.js]: > Do not use this method anymore as it is deprecated.

https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/initMouseEvent

**Comment by NobodysNightmare** [comment]
Closing in favor of https://github.com/opf/openproject/issues/2813
